compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1310,32 +1264,39 @@ impl<'tcx> TyCtxt<'tcx> {
     }
 
     pub fn metadata_encoding_version(self) -> Vec<u8> {
-        self.cstore.metadata_encoding_version().to_vec()
+        self.untracked_resolutions.cstore.metadata_encoding_version().to_vec()
```

Comment:
Can you add comments to this method and `encode_metadata` justifying the use of `untracked_resolutions`, like you did in other methods?